pyiceberg/catalog/rest/auth.py

```diff
@@ -187,3 +217,4 @@ def create(cls, class_or_name: str, config: Dict[str, Any]) -> AuthManager:
 AuthManagerFactory.register("noop", NoopAuthManager)
 AuthManagerFactory.register("basic", BasicAuthManager)
 AuthManagerFactory.register("legacyoauth2", LegacyOAuth2AuthManager)
+AuthManagerFactory.register("google", GoogleAuthManager)
```

Comment:
I can not see any code change to call Google authmanager. This line always create legacyauthmanager: https://github.com/apache/iceberg-python/blob/4cac6910667899238a750adc560f317f62718f82/pyiceberg/catalog/rest/__init__.py#L250

Reply:
Hi @talatuyarer that's right - that is the current behavior.

There's a pending PR that will introduce support for setting different AuthManager implementations here: https://github.com/apache/iceberg-python/pull/2055

Reply:
I'd love to! I'm waiting for a review approval to do so